File: hello.c
#include <stdio.h>

int main(int argc, char *argv[]) {
    if (argc == 1) {
        printf("Hello World\n");
    } else {
        printf("Hello ");
        for (int i = 1; i < argc; i++) {
            printf("%s", argv[i]);
            if (i < argc - 1) {
                printf(" ");
            }
        }
        printf("\n");
    }

    return 0;
}

File: Hellomaven/src/main/java/org/example/Hello.java
package org.example;

public class Hello {
    public static void main(String[] args) {
        if (args.length == 0) {
            System.out.println("Hello, World!");
        } else {
            String separator = "";
            System.out.print("Hello, ");
            for (String name : args) {
                System.out.print(separator + name);
                separator = " ";
            }
            System.out.println("!");
        }
    }
}